tests/sentry/tasks/test_relay.py

```diff
@@ -29,45 +30,6 @@ def _cache_keys_for_org(org):
             yield key.public_key
 
 
-@pytest.fixture
-def emulate_transactions(burst_task_runner, django_capture_on_commit_callbacks):
-    # This contraption helps in testing the usage of `transaction.on_commit` in
-    # schedule_build_project_config. Normally tests involving transactions would
-    # require us to use the transactional testcase (or
-    # `pytest.mark.django_db(transaction=True)`), but that incurs a 2x slowdown
-    # in test speed and we're trying to keep our testcases fast.
```

Comment:
`transaction.on_commit` is now part of the scheduling function (not the caller), so tests no longer have control over creating and emulating transactions.

Reply:
Why does moving the call to `on_commit` change whether or not it can be emulated?

Reply:
It does not change it. My wording could be better, by "control" I wanted to say you were directly using `on_commit` before. I also changed it to make it simpler to read, accepting the test slowdown -- it's fair to push back on that though.

Reply:
I would restore this tbh.. but maybe I'm missing a good reason to switch to transactional test cases.